### PR TITLE
[5.6] Filesystem changes

### PR DESCRIPTION
**Pull request by Smolevich**
Related with [Pull Request](https://github.com/laravel/framework/pull/24438). I add in test code for demonstation problem.

### CODE REVIEW

File: src/Illuminate/Filesystem/Filesystem.php [hunk 1]
```diff
@@ -180,7 +194,7 @@ public function delete($paths)
 
         foreach ($paths as $path) {
             try {
-                if (! @unlink($path)) {
+                if (! unlink($path)) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Filesystem/Filesystem.php]: Why?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Filesystem/Filesystem.php (hunk 1 above)
Comment (Smolevich) [reply src/Illuminate/Filesystem/Filesystem.php]: Because if using supressing errors block catch is never will be executed.

Comment (Smolevich) [reply src/Illuminate/Filesystem/Filesystem.php]: So or use supressing errors or catch ErrorException, not both.

Comment (GrahamCampbell) [reply src/Illuminate/Filesystem/Filesystem.php]: This method returns true or false to indicate success. This cannot be changed.

**Comment by GrahamCampbell** [comment]
Please target 5.7.

File: src/Illuminate/Filesystem/Filesystem.php (hunk 1 above)
Comment (Smolevich) [reply src/Illuminate/Filesystem/Filesystem.php]: What is problem? In laravel all warnings transforms to exception

**Pull request closed without merging**